### PR TITLE
[Migration Review - DO NOT MERGE] Gateway module

### PR DESCRIPTION
**Pull request by red-0ne**
### Human Summary

Review branch for the gateway module migration to rollkit/cosmos-sdk v0.50.x (see: [migration notion doc](https://www.notion.so/buildwithgrove/Cosmos-SDK-v0-50-1-Migration-2d757ec3f88c4251a6712bcb5f3b4e97?pvs=4#556005986ad44a4ab5dab24cc3d997b3)).

Once this branch is approved:

* Double-check that the corresponding merge branch is up to date: `git diff origin/migration/gateway-module..origin/migration/review/gateway-module`
* Merge #371 

NOTE:

`*.pulsar.go` code is auto-generated and can be ignored

## Issue

- #240 

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [x] Other (specify): Module migration to rollkit/cosmos-sdk v0.50.x

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: Makefile [hunk 1]
```diff
@@ -226,12 +237,12 @@ itest: check_go_version ## Run tests iteratively (see usage for more)
 .PHONY: go_mockgen
 go_mockgen: ## Use `mockgen` to generate mocks used for testing purposes of all the modules.
 	find . -name "*_mock.go" | xargs --no-run-if-empty rm
-	go generate ./x/application/types/
+	# go generate ./x/application/types/
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply Makefile:240]: Each module migration is uncommenting its own mock generation. So when they everything is merged they'll be all uncommented.

File: x/gateway/module/config/gateway_config_reader.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package config
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply x/gateway/module/config/gateway_config_reader.go:1]: Checked, there's none that refers to them.

File: x/gateway/module/genesis_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package gateway_test
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply x/gateway/module/genesis_test.go:1]: Good catch! They seem to serve the same purpose. We had them previously on `x/gateway/types/genesis_test.go` but now they seems to belong to `x/gateway/module/genesis_test.go`

File: x/gateway/module/autocli.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package gateway
+
+import (
+	autocliv1 "cosmossdk.io/api/cosmos/autocli/v1"
+
+	modulev1 "github.com/pokt-network/poktroll/api/poktroll/gateway"
+)
+
+// AutoCLIOptions implements the autocli.HasAutoCLIConfig interface.
+func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply x/gateway/module/autocli.go:10]: Currently `autocli` is not used to generate or to handle CLI commands. As of my understanding, we will wire it to our commands when it'll be possible to handle configuration files.

cc @bryanchriswhite

**Comment by red-0ne** [comment]
@Olshansk, I addressed most of the change requests and left a couple of them with comments. PTAL.

**Review by Olshansk — APPROVED** [review]
Reviewed the changes and LGTM.

APproving for reference but DO NOT MERGE

**Comment by bryanchriswhite** [comment]
@red-0ne please add `x/gateway/types.go`:

```
package types

// This file is in place to declare the package for dynamically generated protobufs
```

**Pull request closed without merging**